servlet/src/main/java/io/undertow/servlet/spec/HttpServletRequestImpl.java

```diff
@@ -328,6 +329,11 @@ public String changeSessionId() {
         }
         String newId = underlyingSession.changeSessionId(exchange, originalServletContext.getSessionConfig());
         servletContext.getDeployment().getApplicationListeners().httpSessionIdChanged(session, oldId);
+        //update jsessionid if in form redirect location url
+        String oldLocation = (String) underlyingSession.removeAttribute(ServletFormAuthenticationMechanism.SESSION_KEY);
```

Comment:
This will always remove the attribute, and only add it back if URL rewriting is in use (breaking the use of cookies). I am going to fix this in a different way via the use of SessionListeners, so the logic can be encapsulated in the auth mechanism.